docs-site/content/guide/natural-language-search.md

```diff
@@ -0,0 +1,475 @@
+- model:prius
+- make:[BMW,Nissan] returns cars that are manufactured by BMW OR Nissan.
+
+Numeric Filters: Use :[min..max] for ranges, or comparison operators like :>, :<, :>=, :<=, :=. Examples:
+ - year:[2000..2020]
+ - highway_mpg:>40
+ - msrp:=30000
+
+Multiple Conditions: Separate conditions with &&. Examples:
+ - num_employees:>100 && country:[USA,UK]
+ - categories:=Shoes && categories:=Outdoor
+
+OR Conditions Across Fields: Use || only for different fields. Examples:
+ - vehicle_size:Large || vehicle_style:Wagon
+ - (vehicle_size:Large || vehicle_style:Wagon) && year:>2010
+
+If the same field is used for filtering multiple values in an || (OR) operation, then use the multi-value OR syntax. For eg:
+\`make:BMW || make:Honda || make:Ford\`
+should be simplified as:
+\`make:[BMW, Honda, Ford]\`
+
+Negation: Use :!= to exclude values. Examples:
+ - make:!=Nissan
+ - make:!=[Nissan,BMW]
+
+If any string values have parentheses, surround the value with backticks to escape them.
+
+For eg, if a field has the value "premium unleaded (required)", and you need to use it in a filter_by expression, then you would use it like this:
+
+- fuel_type:\`premium unleaded (required)\`
+- fuel_type!:\`premium unleaded (required)\`
+
+## Sorting (for the sort_by property) ##
+
+You can only sort maximum 3 sort fields at a time. The syntax is {fieldName}: follow by asc (ascending) or dsc (descending), if sort by multiple fields, separate them by a comma. Examples:
+ - msrp:desc
+ - year:asc,city_mpg:desc
+
+Sorting hints:
+  - When a user says something like "good mileage", sort by highway_mpg or/and city_mpg.
+  - When a user says something like "powerful", sort by engine_hp.
+  - When a user says something like "latest", sort by year.
+
+## Car properties ##
+
+| Name | Data Type | Filter | Sort | Enum Values  | Description|
+|------|-----------|--------|------|--------------|------------|
+${await getCachedCollectionProperties()}
+
+### Query (for the query property) ###
+Include query only if both filter_by and sort_by are inadequate.
+
+### User-Supplied Query ###
+
+{{query}}
+
+### Output Instructions ###
+
+Provide the valid JSON with the correct filter and sorting format, only include fields with non-null values. Do not add extra text or explanations.`,
+  )
+```
+
+  </template>
+</Tabs>
+
+### Dynamic Prompt based on the Schema
+
+Notice the `getCachedCollectionProperties()` function in the prompt above.
+
+That function essentially converts the Typesense collection schema into a tabular format with a list of field names and sample enum values in each field. 
+We're using a markdown format to help the LLM recognize these field values in the query and convert them into appropriate field filters.
+
+Here's an example of what the output of that function could look like:
+
+```markdown
+## Car properties ##
+
+| Name | Data Type | Filter | Sort | Enum Values  | Description|
+|------|-----------|--------|------|--------------|------------|
```

Comment:
@phiHero Could you add the actual output of the `getCachedCollectionProperties()` here?